[Bug]: air脚本存在对mac os专用的链接器标志

### 相关组件

后端服务及API

### Bug 描述

在.air.toml中存在标志-no_warn_duplicate_libraries，linux不支持此标志位，导致构建失败
在安装air后make dev-app,构建报错

### 期望行为

能够使linux用户成功允许脚本

### 相关日志

```shell
maomeng@LAPTOP-AHKGEFT9:~/project/WeKnora$ make dev-app
./scripts/dev.sh app
[INFO] 启动后端应用（本地开发模式）...
[INFO] 加载 .env 文件...
[INFO] 环境变量已设置，启动应用...
[INFO] 数据库地址: localhost:5432
[SUCCESS] 检测到 Air，使用热重载模式启动...
[INFO] 修改 Go 代码后将自动重新编译和重启
[warning] build.bin is deprecated; set build.entrypoint instead

  __    _   ___  
 / /\  | | | |_) 
/_/--\ |_| |_| \_ v1.65.1, built with Go go1.25.9

watching .
watching Formula
watching client
watching cmd
watching cmd/desktop
watching cmd/desktop/build
watching cmd/desktop/build/darwin
watching cmd/download
watching cmd/download/duckdb
watching cmd/server
watching config
watching config/prompt_templates
watching dataset
watching dataset/samples
watching deploy
watching docker
watching docker/config
watching docreader
watching docreader/client
watching docreader/models
watching docreader/ocr
watching docreader/parser
watching docreader/proto
watching docreader/scripts
watching docreader/splitter
watching docreader/testdata
watching docreader/testdata/images
watching docreader/utils
!exclude docs
watching examples
watching examples/skills
watching examples/skills/pdf-processing
watching examples/skills/pdf-processing/scripts
!exclude frontend
watching helm
watching helm/templates
watching internal
watching internal/agent
watching internal/agent/memory
watching internal/agent/skills
watching internal/agent/token
watching internal/agent/tools
watching internal/application
watching internal/application/repository
watching internal/application/repository/memory
watching internal/application/repository/memory/neo4j
watching internal/application/repository/retriever
watching internal/application/repository/retriever/elasticsearch
watching internal/application/repository/retriever/elasticsearch/v7
watching internal/application/repository/retriever/elasticsearch/v8
watching internal/application/repository/retriever/milvus
watching internal/application/repository/retriever/neo4j
watching internal/application/repository/retriever/postgres
watching internal/application/repository/retriever/qdrant
watching internal/application/repository/retriever/sqlite
watching internal/application/repository/retriever/weaviate
watching internal/application/service
watching internal/application/service/chat_pipeline
watching internal/application/service/file
watching internal/application/service/llmcontext
watching internal/application/service/memory
watching internal/application/service/metric
watching internal/application/service/retriever
watching internal/assets
watching internal/common
watching internal/config
watching internal/container
watching internal/database
watching internal/datasource
watching internal/datasource/connector
watching internal/datasource/connector/feishu
watching internal/datasource/connector/notion
watching internal/errors
watching internal/event
watching internal/handler
watching internal/handler/session
watching internal/im
watching internal/im/dingtalk
watching internal/im/feishu
watching internal/im/mattermost
watching internal/im/slack
watching internal/im/telegram
watching internal/im/wechat
watching internal/im/wecom
watching internal/infrastructure
watching internal/infrastructure/chunker
watching internal/infrastructure/docparser
watching internal/infrastructure/web_fetch
watching internal/infrastructure/web_search
watching internal/logger
watching internal/mcp
watching internal/middleware
watching internal/models
watching internal/models/asr
watching internal/models/chat
watching internal/models/embedding
watching internal/models/provider
watching internal/models/rerank
watching internal/models/utils
watching internal/models/utils/ollama
watching internal/models/vlm
watching internal/router
watching internal/runtime
watching internal/sandbox
watching internal/searchutil
watching internal/stream
watching internal/tracing
watching internal/types
watching internal/types/interfaces
watching internal/utils
watching mcp-server
!exclude migrations
watching misc
watching packages
watching scripts
watching skills
watching skills/preloaded
watching skills/preloaded/citation-generator
watching skills/preloaded/data-processor
watching skills/preloaded/data-processor/scripts
watching skills/preloaded/doc-coauthoring
watching skills/preloaded/document-analyzer
!exclude testdata
!exclude tmp
building...
# github.com/Tencent/WeKnora/cmd/server
/usr/local/go/pkg/tool/linux_amd64/link: running g++ failed: exit status 1
/usr/bin/g++ -m64 -o $WORK/b001/exe/a.out -rdynamic /tmp/go-link-2567778130/go.o /tmp/go-link-2567778130/000000.o /tmp/go-link-2567778130/000001.o /tmp/go-link-2567778130/000002.o /tmp/go-link-2567778130/000003.o /tmp/go-link-2567778130/000004.o /tmp/go-link-2567778130/000005.o /tmp/go-link-2567778130/000006.o /tmp/go-link-2567778130/000007.o /tmp/go-link-2567778130/000008.o /tmp/go-link-2567778130/000009.o /tmp/go-link-2567778130/000010.o /tmp/go-link-2567778130/000011.o /tmp/go-link-2567778130/000012.o /tmp/go-link-2567778130/000013.o /tmp/go-link-2567778130/000014.o /tmp/go-link-2567778130/000015.o /tmp/go-link-2567778130/000016.o /tmp/go-link-2567778130/000017.o /tmp/go-link-2567778130/000018.o /tmp/go-link-2567778130/000019.o /tmp/go-link-2567778130/000020.o /tmp/go-link-2567778130/000021.o /tmp/go-link-2567778130/000022.o /tmp/go-link-2567778130/000023.o /tmp/go-link-2567778130/000024.o /tmp/go-link-2567778130/000025.o /tmp/go-link-2567778130/000026.o /tmp/go-link-2567778130/000027.o /tmp/go-link-2567778130/000028.o /tmp/go-link-2567778130/000029.o /tmp/go-link-2567778130/000030.o /tmp/go-link-2567778130/000031.o /tmp/go-link-2567778130/000032.o /tmp/go-link-2567778130/000033.o /tmp/go-link-2567778130/000034.o /tmp/go-link-2567778130/000035.o /tmp/go-link-2567778130/000036.o /tmp/go-link-2567778130/000037.o /tmp/go-link-2567778130/000038.o /tmp/go-link-2567778130/000039.o /tmp/go-link-2567778130/000040.o /tmp/go-link-2567778130/000041.o /tmp/go-link-2567778130/000042.o /tmp/go-link-2567778130/000043.o /tmp/go-link-2567778130/000044.o /tmp/go-link-2567778130/000045.o /tmp/go-link-2567778130/000046.o /tmp/go-link-2567778130/000047.o /tmp/go-link-2567778130/000048.o /tmp/go-link-2567778130/000049.o /tmp/go-link-2567778130/000050.o /tmp/go-link-2567778130/000051.o /tmp/go-link-2567778130/000052.o /tmp/go-link-2567778130/000053.o /tmp/go-link-2567778130/000054.o /tmp/go-link-2567778130/000055.o /tmp/go-link-2567778130/000056.o /tmp/go-link-2567778130/000057.o /tmp/go-link-2567778130/000058.o /tmp/go-link-2567778130/000059.o /tmp/go-link-2567778130/000060.o /tmp/go-link-2567778130/000061.o /tmp/go-link-2567778130/000062.o /tmp/go-link-2567778130/000063.o /tmp/go-link-2567778130/000064.o /tmp/go-link-2567778130/000065.o /tmp/go-link-2567778130/000066.o /tmp/go-link-2567778130/000067.o /tmp/go-link-2567778130/000068.o /tmp/go-link-2567778130/000069.o /tmp/go-link-2567778130/000070.o /tmp/go-link-2567778130/000071.o /tmp/go-link-2567778130/000072.o /tmp/go-link-2567778130/000073.o /tmp/go-link-2567778130/000074.o /tmp/go-link-2567778130/000075.o /tmp/go-link-2567778130/000076.o /tmp/go-link-2567778130/000077.o /tmp/go-link-2567778130/000078.o /tmp/go-link-2567778130/000079.o /tmp/go-link-2567778130/000080.o /tmp/go-link-2567778130/000081.o /tmp/go-link-2567778130/000082.o /tmp/go-link-2567778130/000083.o /tmp/go-link-2567778130/000084.o /tmp/go-link-2567778130/000085.o /tmp/go-link-2567778130/000086.o /tmp/go-link-2567778130/000087.o /tmp/go-link-2567778130/000088.o /tmp/go-link-2567778130/000089.o /tmp/go-link-2567778130/000090.o /tmp/go-link-2567778130/000091.o /tmp/go-link-2567778130/000092.o /tmp/go-link-2567778130/000093.o /tmp/go-link-2567778130/000094.o /tmp/go-link-2567778130/000095.o /tmp/go-link-2567778130/000096.o /tmp/go-link-2567778130/000097.o /tmp/go-link-2567778130/000098.o /tmp/go-link-2567778130/000099.o /tmp/go-link-2567778130/000100.o /tmp/go-link-2567778130/000101.o /tmp/go-link-2567778130/000102.o /tmp/go-link-2567778130/000103.o /tmp/go-link-2567778130/000104.o /tmp/go-link-2567778130/000105.o /tmp/go-link-2567778130/000106.o /tmp/go-link-2567778130/000107.o /tmp/go-link-2567778130/000108.o /tmp/go-link-2567778130/000109.o /tmp/go-link-2567778130/000110.o /tmp/go-link-2567778130/000111.o /tmp/go-link-2567778130/000112.o /tmp/go-link-2567778130/000113.o /tmp/go-link-2567778130/000114.o /tmp/go-link-2567778130/000115.o /tmp/go-link-2567778130/000116.o /tmp/go-link-2567778130/000117.o /tmp/go-link-2567778130/000118.o /tmp/go-link-2567778130/000119.o /tmp/go-link-2567778130/000120.o /tmp/go-link-2567778130/000121.o /tmp/go-link-2567778130/000122.o -Wl,-no_warn_duplicate_libraries -lresolv -Wl,-no_warn_duplicate_libraries -lpthread -Wl,-no_warn_duplicate_libraries -lm -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -ldl -Wl,-no_warn_duplicate_libraries -Wl,-no_warn_duplicate_libraries -rdynamic -lduckdb_static -lautocomplete_extension -lcore_functions_extension -licu_extension -ljson_extension -lparquet_extension -ljemalloc_extension -ltpcds_extension -ltpch_extension -lduckdb_fastpforlib -lduckdb_fmt -lduckdb_fsst -lduckdb_hyperloglog -lduckdb_mbedtls -lduckdb_miniz -lduckdb_pg_query -lduckdb_re2 -lduckdb_skiplistlib -lduckdb_utf8proc -lduckdb_yyjson -lduckdb_zstd -lstdc++ -lm -ldl -L/home/maomeng/go/pkg/mod/github.com/duckdb/duckdb-go-bindings/linux-amd64@v0.1.24/ -Wl,-no_warn_duplicate_libraries
/usr/bin/ld: Error: unable to disambiguate: -no_warn_duplicate_libraries (did you mean --no_warn_duplicate_libraries ?)
collect2: error: ld returned 1 exit status

failed to build, error: exit status 1
running...
/bin/sh: 1: /home/maomeng/project/WeKnora/tmp/main: not found
Process Exit with Code: 127
```

### 操作系统

Ubuntu 20.04

### 确认事项

- [x] 我已经搜索了现有的 issues，确认这是一个新问题

--- a/.air.toml
+++ b/.air.toml
@@ -5,7 +5,7 @@ tmp_dir = "tmp"
 [build]
   args_bin = []
   bin = "./tmp/main"
-  cmd = "CGO_CFLAGS='-Wno-deprecated-declarations -Wno-gnu-folding-constant' CGO_LDFLAGS='-Wl,-no_warn_duplicate_libraries' go build -ldflags=\"$(./scripts/get_version.sh ldflags) -X 'google.golang.org/protobuf/reflect/protoregistry.conflictPolicy=warn'\" -o ./tmp/main ./cmd/server"
+  cmd = "CGO_CFLAGS='-Wno-deprecated-declarations -Wno-gnu-folding-constant' $(uname | grep -q Darwin && echo \"CGO_LDFLAGS='-Wl,-no_warn_duplicate_libraries'\") go build -ldflags=\"$(./scripts/get_version.sh ldflags) -X 'google.golang.org/protobuf/reflect/protoregistry.conflictPolicy=warn'\" -o ./tmp/main ./cmd/server"
   delay = 1000
   exclude_dir = ["assets", "tmp", "vendor", "testdata", "frontend", "migrations", "node_modules", "docs"]
   exclude_file = []

